### ISSUE TITLE
`zenMode.hideStatusBar` is not getting applied immediately

### ISSUE THREAD

**Issue by sandy081** [Microsoft/vscode#22280]
- VSCode Version: Insiders
- OS Version: OSX

Steps to Reproduce:

1. Open VS Code
2. Enable Zen mode 
3. Open Settings and set `zenMode.hideStatusBar` to false

Expected: Status bar should be visible
Actual: Status bar is not visible

**Comment by isidorn** [comment]
Yeah, it will be applied only once you re-enter zen mode.
Basically duplicate of https://github.com/Microsoft/vscode/issues/22136
